app/buckets/handlers.js

```diff
@@ -49,25 +49,27 @@ exports.create_bucket = (req, res) => {
 
 
 exports.bucket_details = (req, res, next) => {
-  let bucket;
-  let users;
   Promise.all([Bucket.get(req.params.id), User.list()]) // need to include App.list() in future
-    .then(([returned_bucket, returned_users]) => {
-      bucket = returned_bucket;
-      users = returned_users;
-    })
-    .then(() => bucket.access_logs)
-    .then((access_logs) => { // need to include apps in future
+    .then(([bucket, users]) => { // need to include apps in future
       res.render('buckets/details.html', {
         bucket,
         users_options: users.exclude(bucket.users),
-        access_logs,
       }); // need to include apps_options: apps.exclude(bucket.apps) in future
     })
     .catch(next);
 };
 
 
+exports.access_logs = (req, res, next) => {
+  Bucket.get(req.params.id)
+    .then(bucket => bucket.access_logs(req.query.num_days))
```

Comment:
We should probably restrict `req.query.num_days` to one of the drop-down values, to prevent a user passing an arbitrarily high number for some kind of DoS attack. (The API should do this to, but it's good practice).